packages/rest/src/coercion/coerce-parameter.ts

```diff
@@ -32,7 +32,14 @@ export function coerceParameter(
   data: string | undefined | object,
   spec: ParameterObject,
 ) {
-  const schema = spec.schema;
+  let schema = spec.schema;
+
+  if (!schema && spec.content) {
+    const content = spec.content;
+    const jsonSpec = content['application/json'];
```

Comment:
@deepakrkris The encoding type should be inferred from the request context, `coerceParameter` is called [here](https://github.com/strongloop/loopback-next/blob/4406bd3b9de0fa023896f7c97cb2c98157ca9b71/packages/rest/src/parser.ts#L86), which has access to the `request`.

So I would suggest add a 3rd parameter `options` for `coerceParameter`, and provide the `request` as `options.reqCtx`(or some better name), then retrieve the encoding type from `request`.

If `schema` found then use it, if no `schema` field, search for `spec.content[encodingType]`.

Reply:
@jannyHou , 'application/json' here does not refer to the incoming request's encoding type. It tells what type of content a particular `parameter` carries.  As per Open API spec 3 , `in:query` and `content: application/json: schema` together means the query parameter is a json encoded string.

Reply:
@jannyHou we can safely look for only `application/json` in the content field for condition `in: query` , because Open API v3 supports only this encoding type inside content field for a query parameter.